src/hooks/tradingView/useChartLines.ts

```diff
@@ -1,29 +1,60 @@
 import { useCallback, useEffect, useRef, useState } from 'react';
 
+import { IOrderLineAdapter } from 'public/tradingview/charting_library';
 import { shallowEqual } from 'react-redux';
 
-import { ORDER_SIDES, SubaccountOrder } from '@/constants/abacus';
+import {
+  AbacusOrderType,
+  HumanReadablePlaceOrderPayload,
+  ORDER_SIDES,
+  SubaccountOrder,
+} from '@/constants/abacus';
 import { TOGGLE_ACTIVE_CLASS_NAME } from '@/constants/charts';
+import { DEFAULT_SOMETHING_WENT_WRONG_ERROR_PARAMS } from '@/constants/errors';
 import { STRING_KEYS } from '@/constants/localization';
-import { ORDER_TYPE_STRINGS, type OrderType } from '@/constants/trade';
+import { StatSigFlags } from '@/constants/statsig';
+import { ORDER_TYPE_STRINGS, TradeTypes, type OrderType } from '@/constants/trade';
 import type { ChartLine, PositionLineType, TvWidget } from '@/constants/tvchart';
 
+import {
+  cancelOrderConfirmed,
+  cancelOrderFailed,
+  cancelOrderSubmitted,
+  placeOrderFailed,
+  placeOrderSubmitted,
+  setLatestOrder,
+} from '@/state/account';
 import {
   getCurrentMarketOrders,
   getCurrentMarketPositionData,
   getIsAccountConnected,
 } from '@/state/accountSelectors';
-import { useAppSelector } from '@/state/appTypes';
+import { useAppDispatch, useAppSelector } from '@/state/appTypes';
 import { getAppColorMode, getAppTheme } from '@/state/configsSelectors';
 import { getCurrentMarketId } from '@/state/perpetualsSelectors';
 
+import abacusStateManager from '@/lib/abacus';
 import { MustBigNumber } from '@/lib/numbers';
+import {
+  cancelOrderAsync,
+  createPlaceOrderPayloadFromExistingOrder,
+} from '@/lib/orderModification';
 import { isOrderStatusOpen } from '@/lib/orders';
 import { getChartLineColors } from '@/lib/tradingView/utils';
 
+import { useStatsigGateValue } from '../useStatsig';
 import { useStringGetter } from '../useStringGetter';
 
 const CHART_LINE_FONT = 'bold 10px Satoshi';
+const ORDER_TYPES_MODIFICATION_ENABLED = [
+  AbacusOrderType.StopMarket.ordinal,
+  AbacusOrderType.TakeProfitMarket.ordinal,
+  AbacusOrderType.Limit.ordinal,
+] as number[];
+
+const canModifyOrderTypeFromChart = (order: SubaccountOrder) => {
+  return ORDER_TYPES_MODIFICATION_ENABLED.includes(order.type.ordinal);
+};
```

Comment:
super-nit: consider moving to `lib/tradingView/utils` and `constants/trade`

Reply:
wait or `lib/orderModification`

Reply:
oh yes good idea 😎